Frameworks/Foundation/NSURLResponse.mm

```diff
@@ -89,8 +94,22 @@ - (id)_initWithURL:(NSURL*)url
         _url.attach([url copy]);
         _mimeType.attach([mimeType copy]);
         _textEncodingName.attach([textEncodingName copy]);
-        if (nil != fileName) {
-            _suggestedFilename.attach([fileName copy]);
+
+        if (fileName && ![fileName isEqualToString:s_unknownFileName]) {
+            woc::unique_cf<CFStringRef> identifier {
+                _UTTypeCreatePreferredIdentifierForTag(static_cast<CFStringRef>(@"public.mime-type"),
+                                                       static_cast<CFStringRef>(mimeType),
+                                                       nullptr)
+            };
+            StrongId<NSString> extension {
+                static_cast<NSString*>(
+                    _UTTypeCopyPreferredTagWithClass(identifier.get(), static_cast<CFStringRef>(@"public.filename-extension")))
+            };
```

Comment:
this also masks the fact that it will be adding another +1, leaking this extension.

---
In reply to: [95277280](https://github.com/Microsoft/WinObjC/pull/1662#discussion_r95277280) [](ancestors = 95277280)